File: src/main/java/com/yerti/core/database/mongodb/MongoDBManager.java
package com.yerti.core.database.mongodb;

import com.mongodb.MongoClient;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.model.Filters;
import org.bson.Document;
import org.bson.conversions.Bson;

import java.util.Map;

public class MongoDBManager {
    private MongoCollection<org.bson.Document> players;
    private MongoDatabase mcserverdb;
    private MongoClient client;
    private String database;

    public MongoDBManager(String database) {
        this.database = database;
    }

    public boolean connect(String ip, int port) {
        client = new MongoClient(ip, port);

        mcserverdb = client.getDatabase(database);
        players = mcserverdb.getCollection("players");
        return true;
    }

    public void store(String key, Object param) {
        Document document = new Document(key, param);
        players.insertOne(document);
    }

    public void store(Map<String, Object> params) {
        Document doc = new Document();

        params.forEach(doc::append);

        players.insertOne(doc);
    }

    public void update(String key, Object value, Map<String, Object> params) {
        Document document = new Document();

        params.forEach(document::append);

        Document found = new Document(key, value);

        players.updateOne(found, document);
    }

    public Document get(Bson... filters) {
        return players.find(Filters.and(filters)).first();
    }


}

File: src/main/java/com/yerti/core/database/mongodb/MongoDBManager.kt
package com.yerti.core.database.mongodb

import com.mongodb.MongoClient
import com.mongodb.client.MongoCollection
import com.mongodb.client.MongoDatabase
import com.mongodb.client.model.Filters
import org.bson.Document
import org.bson.conversions.Bson

class MongoDBManager(private val database: String) {
    private var players: MongoCollection<Document>? = null
    private var mcserverdb: MongoDatabase? = null
    private var client: MongoClient? = null
    fun connect(ip: String?, port: Int): Boolean {
        client = MongoClient(ip, port)
        mcserverdb = client!!.getDatabase(database)
        players = mcserverdb!!.getCollection("players")
        return true
    }

    fun store(key: String?, param: Any?) {
        val document = Document(key, param)
        players!!.insertOne(document)
    }

    fun store(params: Map<String?, Any?>) {
        val doc = Document()
        params.forEach { (key: String?, value: Any?) -> doc.append(key, value) }
        players!!.insertOne(doc)
    }

    fun update(key: String?, value: Any?, params: Map<String?, Any?>) {
        val document = Document()
        params.forEach { (key: String?, value: Any?) -> document.append(key, value) }
        val found = Document(key, value)
        players!!.updateOne(found, document)
    }

    operator fun get(vararg filters: Bson?): Document {
        return players!!.find(Filters.and(*filters)).first()
    }

}